logging/telemetry.go

```diff
@@ -93,24 +93,44 @@ func makeTelemetryState(cfg TelemetryConfig, hookFactory hookFactory) (*telemetr
 func ReadTelemetryConfigOrDefault(dataDir string, genesisID string) (cfg TelemetryConfig, err error) {
 	err = nil
 	dataDirProvided := dataDir != ""
+	var configPath string
+
+	// If we have a data directory, then load the config
 	if dataDirProvided {
-		configPath := filepath.Join(dataDir, TelemetryConfigFilename)
+		configPath = filepath.Join(dataDir, TelemetryConfigFilename)
+		// Load the config, if the GUID is there then we are all set
+		// However if it isn't there then we must create it, save the file and load it.
 		cfg, err = LoadTelemetryConfig(configPath)
 	}
+
+	// We couldn't load the telemetry config for some reason
+	// If the reason is because the directory doesn't exist or we didn't provide a data directory then...
 	if (err != nil && os.IsNotExist(err)) || !dataDirProvided {
-		var configPath string
+
 		configPath, err = config.GetConfigFilePath(TelemetryConfigFilename)
 		if err != nil {
+			// In this case we don't know what to do since we couldn't
+			// create the directory.  Just create an ephemeral config.
 			cfg = createTelemetryConfig()
 			return
 		}
+
+		// Load the telemetry from the default config path
 		cfg, err = LoadTelemetryConfig(configPath)
 	}
+
+	// If there was some error loading the configuration from the config path...
 	if err != nil {
+		// Create an ephemeral config
 		cfg = createTelemetryConfig()
+
+		// If the error was that the the config wasn't there then it wasn't really an error
+		// We actually want to save the information here
 		if os.IsNotExist(err) {
+			cfg.Save(configPath)
```

Comment:
I think we can leave out the save, it's possible this was left out by design. It looks like `diagcfg` is the only consumer of this function and it saves the config itself after making a modification, maybe move this function into the diagcfg package?

Reply:
With the save, what would make this GUID static ?

Reply:
Diagcfg uses this function and saves the config file immediately after fetching the default. Algod/Algoh use a different `EnsureTelemetry` function which does the create/save or read.

Reply:
The save makes it so that the ephemeral config created on 125 is saved and loaded the next time this function (or the related EnsureTelemetryConfigCreated) is called.

Reply:
> I think we can leave out the save, it's possible this was left out by design. It looks like `diagcfg` is the only consumer of this function and it saves the config itself after making a modification, maybe move this function into the diagcfg package?

It does seem to only be called there.  I'm fine dropping it if everyone agrees.